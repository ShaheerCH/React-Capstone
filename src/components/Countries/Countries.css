.flag__section img {
  border-radius: 5px 5px 0 0;
}

.details {
  border-radius: 0 0 5px 5px;
}

h2 {
  padding-bottom: 10px;
  font-size: 20px;
  font-weight: 800;
}

h3 {
  padding-bottom: 5px;
  font-size: 16px;
  font-weight: 600;
}

h3 > span {
  color: #5983fc;
}
