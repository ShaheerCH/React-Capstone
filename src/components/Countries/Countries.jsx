import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import { getAllCountries, fetchAsyncCountries } from '../../features/CountriesSlice';
import { useSelector, useDispatch } from 'react-redux';
import './Countries.css'


const Countries = () => {

  const dispatch = useDispatch()

  useEffect(() => {
    dispatch(fetchAsyncCountries())
  }, [])

  const countries = useSelector(getAllCountries)
  console.log(countries);

  return (
    <section className="grid">
      {
        countries.map((country) => {
          return (
            <Link to={`/countries/${country.name.common}`} key={country.name.common}>
              <article>
                <div className="flag__section">
                  <img src={country.flags.svg} alt="" />
                </div>

                <div className="details">
                  <h2>{country.name.common}</h2>
                  <h3>Population: <span>{country.population}</span></h3>
                  <h3>Region: <span>{country.region}</span></h3>
                  <h3>Capital: <span>{country.capital}</span></h3>
                </div>
              </article>
            </Link>
          )
        })
      }
    </section>
  )
}

export default Countries