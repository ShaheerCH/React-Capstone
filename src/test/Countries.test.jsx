import React from 'react';
import { Provider } from 'react-redux';
import store from '../features/configureStore';
import { MemoryRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { expect, it } from 'vitest';
import Countries from '../components/Countries/Countries';

it('Renders the App Component', () => {
  const CountriesRenderer = renderer.create(
    <Provider store={store}>
      <Countries />
    </Provider>
  ).toJSON();
  expect(CountriesRenderer).toMatchSnapshot()
})