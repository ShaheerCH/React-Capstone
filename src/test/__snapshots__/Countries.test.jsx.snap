// Vitest Snapshot v1

exports[`Renders the App Component 1`] = `
<section
  className="grid"
/>
`;
